js/utilities.js

```diff
@@ -436,6 +437,101 @@ var ERMrest = (function(module) {
             typeof element.descending == 'boolean');
     };
 
+    /*
+     * @function
+     * @private
+     * @param {Object} md The markdown-it object
+     * @param {Object} md The markdown-it-container object.
+     * @desc Sets functionality for custom markdown tags like `iframe` using `markdown-it-container` plugin.
+     */
+    module._bindCustomMarkdownTags = function(md, mdContainer) {
+
+        // Dependent on 'markdown-it-container' and 'markdown-it-attrs' plugins
+        md.use(mdContainer, 'iframe', {
+            /*
+             * Checks whether string matches format "::: iframe [CAPTION](LINK){ATTR=VALUE .CLASSNAME}"
+             * String inside '{}' is Optional, specifies attributes to be applied to prev element
+             */ 
+            validate: function(params) {
+                return params.trim().match(/^iframe\s+(\[[^\]]*\]\([^\[\]]*?\)(\{.*\})?)$/i);
+            },
+
+            render: function (tokens, idx) {
+                // Get token string after regeexp matching to determine actual internal markdown 
+                var m = tokens[idx].info.trim().match(/^iframe\s+(\[[^\]]*\]\([^\[\]]*?\)(\{.*\})?)$/i);
+
+                // If this is the opening tag i.e. starts with "::: iframe " 
+                if (tokens[idx].nesting === 1 && m.length > 0) {
+
+                    // Extract remaining string before closing tag and get its parsed markdown attributes
+                    var attrs = md.parseInline(m[1]), html = "";
+
+                    if (attrs && attrs.length == 1 && attrs[0].children) { 
+
+                        // Check If the markdown is a link
+                        if (attrs[0].children[0].type == "link_open") {
+                            var iframeHTML = "<iframe ", openingLink = attrs[0].children[0];
+
+                            // Add all attributes to the iframe
+                            openingLink.attrs.forEach(function(attr) {
+                                if (attr[0] == "href") {
+                                    iframeHTML += 'src="' + attr[1] + '"';
+                                } else {
+                                    iframeHTML +=  attr[0] + '="' + attr[1] + '"';
+                                }
+                               iframeHTML += " ";
+                            });
+                            html += iframeHTML + "></iframe>";
+
+                            // If there is a caption then add it as a "div" with "caption" class
+                            if (attrs[0].children[1].type == "text") {
+                               html = '<div class="caption">' + md.renderInline(attrs[0].children[1].content)  + "</div>" + html;
+                            }
+
+                            // Encapsulate the iframe inside a div
+                            html = '<p>' + html + "</p>";
+                        }  
+                    }
```

Comment:
This doesn't feel right.  If the conditions are false, we return an empty HTML string?  Shouldn't we drop the iframe container and render the inner markdown if we don't know what to do with it?